File: API_DBFruits/API_DBFruits/Models/DonHang.cs
using System;
using System.Collections.Generic;

namespace API_DBFruits.Models;

public partial class DonHang
{
    public int DonHangId { get; set; }

    public string? KhachHangId { get; set; }

    public string? DiaChi { get; set; }

    public DateOnly NgayDat { get; set; }

    public int? TongTien { get; set; }

    public string? TrangThai { get; set; }

    public virtual ICollection<ChiTietDonHang> ChiTietDonHangs { get; set; } = new List<ChiTietDonHang>();

    public virtual User? KhachHang { get; set; }
}

File: lib/models/DonHang.dart
class DonHang {
  DonHang({
    required this.donHangId,
    required this.khachHangId,
    required this.diaChi,
    required this.ngayDat,
    required this.tongTien,
    required this.trangThai,
  });

  final int? donHangId;
  final String? khachHangId;
  final DateTime? ngayDat;
  final String? diaChi;
  final int? tongTien;
  final String? trangThai;

  factory DonHang.fromJson(Map<String, dynamic> json) {
    return DonHang(
      donHangId: json["donHangId"],
      khachHangId: json["khachHangId"],
      diaChi: json["diaChi"],
      ngayDat: DateTime.tryParse(json["ngayDat"] ?? ""),
      tongTien: json["tongTien"],
      trangThai: json["trangThai"],
    );
  }
}
